### PR TITLE
Calculate string length in chunk_is_str instead of manually specifying it.

### PR DESCRIPTION
**Pull request by dundargoc**
This will prevent any length-counting errors since it won't be done
manually anymore.

### CODE REVIEW

File: src/chunk.h [hunk 1]
```diff
@@ -985,11 +985,13 @@ static inline bool chunk_is_type(Chunk *pc)
 }
 
 
-static inline bool chunk_is_str(Chunk *pc, const char *str, size_t len)
+static inline bool chunk_is_str(Chunk *pc, const string &str)
```
Comment (micheleCTDE) [inline src/chunk.h:988]: We can keep using "const char *" for the string, so we avoid unnecessary conversion to string objects.
We can determine the length on the string with strlen.

File: src/chunk.h [hunk 2]
```diff
@@ -985,11 +985,13 @@ static inline bool chunk_is_type(Chunk *pc)
...
 {
-   return(  pc != nullptr                         // valid pc pointer
-         && (pc->Len() == len)                    // token size equals size parameter
-         && (memcmp(pc->Text(), str, len) == 0)); // token name is the same as str parameter
+   auto len = str.size();
```
Comment (micheleCTDE) [inline src/chunk.h:990]: Can use strlen on a const char pointer.
Possibly avoid using "auto": although it is easier, it makes the code less readable (one has to think about the type...)

**Review by micheleCTDE — CHANGES_REQUESTED** [review]
The idea is good, please amend as per comments below.

File: src/chunk.h (hunk 2 above)
Comment (dundargoc) [reply src/chunk.h]: Good point.

File: src/chunk.h (hunk 1 above)
Comment (dundargoc) [reply src/chunk.h]: Sure. I'll need to adjust the commit message.

Comment (dundargoc) [reply src/chunk.h]: In hindsight, maybe we should just use a string and get rid of the `memcmp` which is just a goofy C-ism at this point?

Comment (micheleCTDE) [reply src/chunk.h]: There is nothing wrong in using C functions.... definitely nothing goofy in it :-) We could perhaps replace ```memcmp``` with strncmp, I missed that on the frist review, but either way would work fine.

**Review by micheleCTDEAdmin — APPROVED** [review]
Looks good now, thanks for updating as requested

**Review by micheleCTDE — APPROVED** [review]

File: src/chunk.h (hunk 1 above)
Comment (micheleCTDE) [reply src/chunk.h]: neovim rocks btw :wink:

**Pull request merged**